prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -536,12 +554,34 @@ void rxr_pkt_handle_data_copied(struct rxr_ep *ep,
 				size_t data_size)
 {
 	struct rxr_rx_entry *rx_entry;
+	ssize_t ret;
 
 	rx_entry = pkt_entry->x_entry;
 	assert(rx_entry);
 	rx_entry->bytes_copied += data_size;
 
 	if (rx_entry->total_len == rx_entry->bytes_copied) {
+		if (rx_entry->fi_flags & RXR_DELIVERY_COMPLETE_REQUESTED) {
+			ret = rxr_pkt_post_ctrl_or_queue(ep,
+							 RXR_RX_ENTRY,
+							 rx_entry,
+							 RXR_RECEIPT_PKT, 0);
+			if (OFI_UNLIKELY(ret)) {
+				FI_WARN(&rxr_prov,
+					FI_LOG_CQ,
+					"Posting of receipt packet failed! err=%s\n",
+					strerror(ret));
+				efa_eq_write_error(&ep->util_ep,
+						   FI_EIO,
+						   ret);
+				rxr_release_rx_entry(ep,
+						     rx_entry);
+			}
+			rxr_cq_handle_rx_completion(ep, pkt_entry, rx_entry);
+			rxr_msg_multi_recv_free_posted_entry(ep, rx_entry);
+			rxr_release_rx_entry(ep, rx_entry);
```

Comment:
I think we discussed this, ```rxr_release_rx_entry``` need to be called in ```rxr_pkt_handle_receipt_sent```, because rxr_pkt_post_ctrl_or_queue() does not always sent the packet successfully, and the construction of receipt packet need rx_entry.

Reply:
Yeah. But then we need to define a new member in `struct rxr_rx_entry`. Because we have to call `rxr_cq_handle_rx_completion ` before calling `rxr_release_rx_entry ` in `rxr_pkt_handle_receipt_sent `. And we need a pointer to the `pkt_entry` that will be passed to `rxr_cq_handle_rx_completion` so as to retrieve it in `rxr_pkt_handle_receipt_sent`.

Reply:
We can still call ```rxr_cq_handle_rx_completion``` here because data has been received. we just cannot release rx_entry here.

Reply:
Currently `rxr_cq_handle_rx_completion` is called here while `rxr_release_rx_entry` is not, resolving.